core/nodejsActionBase/launcher.js

```diff
@@ -15,6 +15,8 @@
  * limitations under the License.
  */
 
+const isLambda = require('./mainFunctionDetection').isLambda;
```

Comment:
Let's make sure we actually test this. I think the compile step copies this file into the user's code directory so a relative resolution might not work. The logshipping test should catch this.

Reply:
Do you think we could add an automated test for the log shipping case? I could test log shipping with this manually but I don't like the idea of us adding features over time without having every case automatic tested.

Reply:
There's multiple ways this could be automatically tested, yes. In the E2E test framework for runtimes, it might be viable to pass an env var to the runtime at container startup. We could override the `OW_INIT_IN_ACTIONLOOP` variable there to be empty to force us to go through the "old" launcher.

Reply:
We can continue to use relative paths but then we'll have to copy the new file you created in the compile step as well.

Reply:
Relative paths didn't work well. It worked fine for when prelauncher.js was used as the entry point. But it didn't work when launcher.js was used as the entry point (by using that "init in ActionLoop" env var to force it). I don't understand why, because my change in serverless/main made it so that the `makeNode` function there copies in the new JS file. But absolute paths made it work always, so I just used absolute paths.

Reply:
As mentioned above, it didn't work because when the user action is initialized, we copy the `launcher.js` file to **their** code's directory (in `/tmp` somewhere).